.jules/bolt.md

```
@@ -5,3 +5,11 @@
 ## 2025-05-23 - [reqwest::Client Cloning Efficiency]
 **Learning:** Cloning a `reqwest::Client` (and its blocking counterpart) is a very cheap operation because they internally use `Arc` to share the connection pool. This makes it efficient to pass clients by value or store them in multiple structs without losing the benefits of connection pooling.
 **Action:** When using shared clients, it is perfectly fine to clone them for use in different components or tasks, as long as they originate from the same initial instance to maintain the shared connection pool.
+
+## 2025-05-24 - [Pre-compiling Regex and Selectors]
+**Learning:** Recompiling  or  in a loop or a frequently called function is a major performance bottleneck. Compiling them once and storing them in a  variable with  can lead to massive speedups (e.g., 400x+).
+**Action:** Always pre-compile regexes and CSS selectors used in scrapers or hot paths. Use  for lazy, thread-safe initialization.
+
+## 2025-05-24 - [Pre-compiling Regex and Selectors]
```

markdownlint-cli2 (0.21.0)

[warning] Multiple headings with the same content

(MD024, no-duplicate-heading)